core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -742,7 +742,7 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
         longArray.set(pos * 2 + 1, keyHashcode);
         isDefined = true;
 
-        if (numKeys > growthThreshold && longArray.size() < MAX_CAPACITY) {
+        if (numKeys >= growthThreshold && longArray.size() < MAX_CAPACITY) {
```

Comment:
The re-allocated space might not be used if no further insertion. Shall we do `growAndRehash` at the beginning of `append` when `numKeys == growthThreshold && !isDefined`?